posthog/hogql/database/schema/util/where_clause_extractor.py

```diff
@@ -266,44 +266,44 @@ def handle_timestamp_comparison(
                 return ast.And(
                     exprs=[
                         ast.CompareOperation(
-                            op=ast.CompareOperationOp.LtEq,
-                            left=ast.ArithmeticOperation(
+                            op=ast.CompareOperationOp.GtEq,
+                            left=rewrite_timestamp_field(node.left, self.timestamp_field, self.context),
+                            right=ast.ArithmeticOperation(
                                 op=ast.ArithmeticOperationOp.Sub,
-                                left=rewrite_timestamp_field(node.left, self.timestamp_field, self.context),
+                                left=node.right,
                                 right=self.time_buffer,
                             ),
-                            right=node.right,
                         ),
                         ast.CompareOperation(
-                            op=ast.CompareOperationOp.GtEq,
-                            left=ast.ArithmeticOperation(
+                            op=ast.CompareOperationOp.LtEq,
+                            left=rewrite_timestamp_field(node.left, self.timestamp_field, self.context),
+                            right=ast.ArithmeticOperation(
                                 op=ast.ArithmeticOperationOp.Add,
-                                left=rewrite_timestamp_field(node.left, self.timestamp_field, self.context),
+                                left=node.right,
                                 right=self.time_buffer,
                             ),
-                            right=node.right,
                         ),
```

Comment:
I flipped the order of these two expressions because I normally think of these as lower and upper bounds conditions and expect to read them in that respective order, and assume others do also (but that may very well just be a quirk of how my brain functions)